community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_1/TransactionBoundQueryContext.scala

```diff
@@ -583,26 +584,60 @@ final class TransactionBoundQueryContext(val transactionalContext: Transactional
     pathFinder.findAllPaths(left, right).iterator().asScala
   }
 
-  override def callReadOnlyProcedure(name: QualifiedProcedureName, args: Seq[Any]) =
-    callProcedure(name, args, transactionalContext.statement.readOperations().procedureCallRead)
+  override def callReadOnlyProcedure(name: QualifiedProcedureName, args: Seq[Any], allowed: String) = {
+    val revertable = transactionalContext.accessMode match {
+      case a: AuthSubject if a.hasRole(allowed) =>
```

Comment:
I thought we would have a comma separated list of roles to be allowed to give better usability, but I realize that it was not specified on the card.